compiler/ballerina-lang/src/main/java/io/ballerina/projects/Module.java

```diff
@@ -120,4 +135,114 @@ public Spliterator spliterator() {
             return this.documentList.spliterator();
         }
     }
+
+    /**
+     * Inner class that handles module modifications.
+     */
+    public static class Modifier {
+        private ModuleId moduleId;
+        private ModuleName moduleName;
+        private Map<DocumentId, DocumentContext> srcDocContextMap;
+        private Map<DocumentId, DocumentContext> testDocContextMap;
+        private boolean isDefaultModule;
+        private Set<ModuleDependency> moduleDependencies;
+        private Package packageInstance;
+        private Project project;
+
+
+        private Modifier(Module oldModule) {
+            moduleId = oldModule.moduleId();
+            moduleName = oldModule.moduleName();
+            srcDocContextMap = copySrcDocs(oldModule);
+            testDocContextMap = copyTestDocs(oldModule);
+            isDefaultModule = oldModule.isDefaultModule();
+            moduleDependencies = new HashSet<>(oldModule.moduleDependencies());
+            packageInstance = oldModule.packageInstance;
+            project = oldModule.project();
+        }
+
+        Modifier updateDocument(DocumentContext newDocContext) {
+            if (this.srcDocContextMap.get(newDocContext.documentId()) != null) {
```

Comment:
If we do not allow null values in the map, let's use the `Map.containsKey` API

Reply:
FixedFixed in 0d89c8f